### ISSUE TITLE
exporter-auto POM file is broken since version 0.18.1-alpha

### ISSUE THREAD

**Issue by ViliusS** [GoogleCloudPlatform/opentelemetry-operations-java#185]
exporter-auto POM file has `<dependencies>` section empty in the release since 0.18.1-alpha. This makes it impossible to use it along Google Cloud libraries BOM to resolve dependencies correctly.

**Comment by aabmass** [comment]
@ViliusS can you add some more information? I don't think we include a POM file in this repo, but the gradle file does define dependencies https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/60cf3d675eda0900258853d568c21bfd4d692d49/exporters/auto/build.gradle#L23

**Comment by ViliusS** [comment]
Check this POM https://search.maven.org/artifact/com.google.cloud.opentelemetry/exporter-auto/0.23.0-alpha/jar and compare it to https://search.maven.org/artifact/com.google.cloud.opentelemetry/exporter-auto/0.18.0-alpha/pom

So, if you want to use exporter-auto with Google Cloud libraries BOM, you also have to include exporter-metrics or exporter-trace so that BOM will resolve all dependencies correctly. In short with 0.18.1-alpha it is enough to have:

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.google.cloud</groupId>
                <artifactId>libraries-bom</artifactId>
                <version>${google-cloud-libraries.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
            <dependency>
                <groupId>io.opentelemetry.instrumentation</groupId>
                <artifactId>opentelemetry-instrumentation-bom-alpha</artifactId>
                <version>${opentelemetry-instrumentation.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

       <dependency>
            <groupId>com.google.cloud.opentelemetry</groupId>
            <artifactId>exporter-auto</artifactId>
            <version>0.18.1-alpha</version>
        </dependency>

But for 0.23.0-alpha you have to do:
```
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.google.cloud</groupId>
                <artifactId>libraries-bom</artifactId>
                <version>${google-cloud-libraries.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
            <dependency>
                <groupId>io.opentelemetry.instrumentation</groupId>
                <artifactId>opentelemetry-instrumentation-bom-alpha</artifactId>
                <version>${opentelemetry-instrumentation.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

       <dependency>
            <groupId>com.google.cloud.opentelemetry</groupId>
            <artifactId>exporter-auto</artifactId>
            <version>0.23.0-alpha</version>
        </dependency>
       <dependency>
            <groupId>com.google.cloud.opentelemetry</groupId>
            <artifactId>exporter-trace</artifactId>
            <version>0.23.0</version>
        </dependency>
```
Which will resolve to newer `grpc-google-cloud-trace-v2` and friends. as needed.

**Comment by dashpole** [comment]
@psx95 can you take a look?

**Comment by psx95** [comment]
Hi @ViliusS, the missing dependencies block starting from v0.18.1-alpha is expected since we started publishing shaded JARs for `exporter-auto` - this means that the `exporter-auto` should be sufficient by itself and you would not need to include `exporter-trace` or `exporter-metrics` separately. 

I ran our [autoinstrument example](https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/tree/60cf3d675eda0900258853d568c21bfd4d692d49/examples/autoinstrument) which uses `exporter-auto` after modifying its dependency block to use the google-cloud-bom, but it ran fine without any issues.

```
dependencies {
	agent agentLibraries.agent
	agent implementation("com.google.cloud.opentelemetry:exporter-auto:0.24.0-alpha") // modified dependency

	implementation(libraries.spring_boot_starter_web)
	implementation(libraries.spring_cloud_starter_openfeign)
	// Temporary, for log-trace correlation
	implementation(libraries.opentelemetry_api)
	implementation('com.google.cloud:libraries-bom:26.10.0') // modified dependency to use BOM
}
```

I was able to generate traces in my Google Cloud project using this example.

Can you provide us with the steps that you took / sample code that you wrote which lead to an error or any other issues you maybe facing ?

**Comment by ViliusS** [comment]
@psx95 let me revise a little bit. I needed BOM to solve conflict with 3rd party dependency described here https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/issues/184. That particular conflict is solved now, but I think there could be other situations where you need to have Google Libraries BOM to solve conflicts.

Hence this ticket should be looked at not from the perspective of "does auto instrumentation work" but from the context of "how and what dependencies are resolved when using exporter-auto with 3rd party BOMs". 

For example, both cases below which use 0.18.0 and 0.23.0 works, but they resolve to different dependency tree. In particular, `grpc` in first example is updated to latest version from Google BOM, but in the second it example it is very old. The only way to have dependency tree more or less the same is to have `export-trace` added to the second example.

```
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>com.google.cloud</groupId>
            <artifactId>libraries-bom</artifactId>
            <version>${google-cloud-libraries.version}</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry.instrumentation</groupId>
            <artifactId>opentelemetry-instrumentation-bom-alpha</artifactId>
            <version>${opentelemetry-instrumentation.version}</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
    </dependencies>
</dependencyManagement>

   <dependency>
        <groupId>com.google.cloud.opentelemetry</groupId>
        <artifactId>exporter-auto</artifactId>
        <version>0.18.1-alpha</version>
    </dependency>
```
```
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.google.cloud</groupId>
                <artifactId>libraries-bom</artifactId>
                <version>${google-cloud-libraries.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
            <dependency>
                <groupId>io.opentelemetry.instrumentation</groupId>
                <artifactId>opentelemetry-instrumentation-bom-alpha</artifactId>
                <version>${opentelemetry-instrumentation.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

       <dependency>
            <groupId>com.google.cloud.opentelemetry</groupId>
            <artifactId>exporter-auto</artifactId>
            <version>0.23.0-alpha</version>
        </dependency>
```

I don't know if that's only Maven thing, or is it the same with Gradle, though, You should probably test with Maven to be sure.

**Comment by ViliusS** [comment]
> ```
> dependencies {
> 	agent agentLibraries.agent
> 	agent implementation("com.google.cloud.opentelemetry:exporter-auto:0.24.0-alpha") // modified dependency
> 
> 	implementation(libraries.spring_boot_starter_web)
> 	implementation(libraries.spring_cloud_starter_openfeign)
> 	// Temporary, for log-trace correlation
> 	implementation(libraries.opentelemetry_api)
> 	implementation('com.google.cloud:libraries-bom:26.10.0') // modified dependency to use BOM
> }
> ```

It looks like on Gradle you need to use `implementation platform` instead of `implementation` to include BOM.

**Comment by psx95** [comment]
@ViliusS, I am looking into this. I will try it out with Maven once as you suggested, but as per your question on "how and what dependencies are resolved when using exporter-auto with 3rd party BOMs", I think `exporter-auto` should be treated as an artifact which does not have any transitive dependency. Since currently its provided as a shaded artifact it bundles every single dependency in itself. 

That being said, I noticed that we are not moving the classes to a shaded package for `exporter-auto` right now which can still lead to issues described in #161 (*Is this something you are running into ?*). We are moving shaded classes for our other resourcemapping module [here](https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/main/shared/resourcemapping/build.gradle#L36). 

As for dependency resolution with `exporter-auto` with 3rd party BOMs, I did some experiments on gradle for now - 

Here is the dependency section (Thank you for pointing out the missing `platform`) - 

```
    // exporter-trace should not be needed when using exporter-auto, since it bundles it. 
    // its added here just to show dependency resolution
    implementation 'com.google.cloud.opentelemetry:exporter-trace:0.24.0' 
    implementation 'com.google.cloud.opentelemetry:exporter-auto:0.24.0-alpha'

    // gRPC
    runtimeOnly 'io.grpc:grpc-netty-shaded:1.52.1'
    implementation 'io.grpc:grpc-protobuf:1.52.1'
    implementation 'io.grpc:grpc-stub:1.52.1'
    compileOnly 'org.apache.tomcat:annotations-api:6.0.53' // necessary for Java 9+

    // BOMs
    implementation(platform('com.google.cloud:libraries-bom:26.10.0'))
    implementation(platform('io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.23.0-alpha'))
```

I analyzed the dependencies using gradle's in-built dependency tool. 

```
gradle -q dependencies --configuration runtimeClasspath
```

Which gave me the following output - 
```
------------------------------------------------------------
Root project 'TestProject'
------------------------------------------------------------

runtimeClasspath - Runtime classpath of source set 'main'.
+--- com.google.cloud.opentelemetry:exporter-trace:0.24.0
|    +--- io.opentelemetry:opentelemetry-semconv:1.15.0-alpha -> 1.23.1-alpha
|    |    \--- io.opentelemetry:opentelemetry-api:1.23.1
|    |         \--- io.opentelemetry:opentelemetry-context:1.23.1
|    +--- com.google.cloud.opentelemetry:shared-resourcemapping:0.24.0
|    +--- com.google.auto.value:auto-value-annotations:1.10.1
|    +--- io.opentelemetry:opentelemetry-api:1.15.0 -> 1.23.1 (*)
|    +--- io.opentelemetry:opentelemetry-sdk:1.15.0 -> 1.23.1
|    |    +--- io.opentelemetry:opentelemetry-sdk-logs:1.23.1-alpha
|    |    |    +--- io.opentelemetry:opentelemetry-api-logs:1.23.1-alpha
|    |    |    |    \--- io.opentelemetry:opentelemetry-api:1.23.1 (*)
|    |    |    +--- io.opentelemetry:opentelemetry-api-events:1.23.1-alpha
|    |    |    |    \--- io.opentelemetry:opentelemetry-api:1.23.1 (*)
|    |    |    \--- io.opentelemetry:opentelemetry-sdk-common:1.23.1
|    |    |         +--- io.opentelemetry:opentelemetry-semconv:1.23.1-alpha (*)
|    |    |         \--- io.opentelemetry:opentelemetry-api:1.23.1 (*)
|    |    +--- io.opentelemetry:opentelemetry-api:1.23.1 (*)
|    |    +--- io.opentelemetry:opentelemetry-sdk-common:1.23.1 (*)
|    |    +--- io.opentelemetry:opentelemetry-sdk-trace:1.23.1
|    |    |    +--- io.opentelemetry:opentelemetry-semconv:1.23.1-alpha (*)
|    |    |    +--- io.opentelemetry:opentelemetry-api:1.23.1 (*)
|    |    |    \--- io.opentelemetry:opentelemetry-sdk-common:1.23.1 (*)
|    |    \--- io.opentelemetry:opentelemetry-sdk-metrics:1.23.1
|    |         +--- io.opentelemetry:opentelemetry-api:1.23.1 (*)
|    |         \--- io.opentelemetry:opentelemetry-sdk-common:1.23.1 (*)
|    +--- com.google.cloud:google-cloud-core:2.8.28 -> 2.12.0
|    |    +--- com.google.guava:guava:31.1-jre
|    |    |    +--- com.google.guava:failureaccess:1.0.1
|    |    |    +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|    |    |    +--- com.google.code.findbugs:jsr305:3.0.2
|    |    |    +--- org.checkerframework:checker-qual:3.12.0 -> 3.31.0
|    |    |    +--- com.google.errorprone:error_prone_annotations:2.11.0 -> 2.18.0
|    |    |    \--- com.google.j2objc:j2objc-annotations:1.3
|    |    +--- com.google.guava:failureaccess:1.0.1
|    |    +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|    |    +--- org.checkerframework:checker-qual:3.31.0
|    |    +--- com.google.errorprone:error_prone_annotations:2.18.0
|    |    +--- com.google.j2objc:j2objc-annotations:1.3
|    |    +--- com.google.api:gax:2.23.2
|    |    +--- io.opencensus:opencensus-api:0.31.1
|    |    +--- io.grpc:grpc-context:1.53.0
|    |    +--- com.google.auto.value:auto-value-annotations:1.10.1
|    |    +--- com.google.protobuf:protobuf-java-util:3.21.12
|    |    +--- com.google.code.gson:gson:2.10.1
|    |    +--- com.google.api.grpc:proto-google-common-protos:2.14.2
|    |    |    \--- com.google.protobuf:protobuf-java:3.21.12
|    |    +--- com.google.api.grpc:proto-google-iam-v1:1.9.2
|    |    +--- org.threeten:threetenbp:1.6.5
|    |    +--- com.google.api:api-common:2.6.2
|    |    +--- com.google.auto.value:auto-value:1.10.1
|    |    +--- javax.annotation:javax.annotation-api:1.3.2
|    |    +--- com.google.auth:google-auth-library-credentials:1.16.0
|    |    +--- com.google.auth:google-auth-library-oauth2-http:1.16.0
|    |    +--- com.google.http-client:google-http-client:1.43.0
|    |    +--- org.apache.httpcomponents:httpclient:4.5.14
|    |    +--- commons-logging:commons-logging:1.2
|    |    +--- commons-codec:commons-codec:1.15
|    |    +--- org.apache.httpcomponents:httpcore:4.4.16
|    |    +--- io.opencensus:opencensus-contrib-http-util:0.31.1
|    |    +--- com.google.http-client:google-http-client-gson:1.43.0
|    |    +--- com.google.protobuf:protobuf-java:3.21.12
|    |    \--- com.google.code.findbugs:jsr305:3.0.2
|    +--- com.google.cloud:google-cloud-trace:2.1.12 -> 2.12.0
|    |    +--- io.grpc:grpc-api:1.53.0
|    |    |    +--- io.grpc:grpc-context:1.53.0
|    |    |    +--- com.google.code.findbugs:jsr305:3.0.2
|    |    |    +--- com.google.errorprone:error_prone_annotations:2.14.0 -> 2.18.0
|    |    |    \--- com.google.guava:guava:31.1-android -> 31.1-jre (*)
|    |    +--- io.grpc:grpc-context:1.53.0
|    |    +--- com.google.code.findbugs:jsr305:3.0.2
|    |    +--- com.google.errorprone:error_prone_annotations:2.18.0
|    |    +--- io.grpc:grpc-stub:1.53.0
|    |    |    +--- io.grpc:grpc-api:1.53.0 (*)
|    |    |    +--- com.google.guava:guava:31.1-android -> 31.1-jre (*)
|    |    |    \--- com.google.errorprone:error_prone_annotations:2.14.0 -> 2.18.0
|    |    +--- io.grpc:grpc-protobuf:1.53.0
|    |    |    +--- io.grpc:grpc-api:1.53.0 (*)
|    |    |    +--- com.google.code.findbugs:jsr305:3.0.2
|    |    |    +--- com.google.protobuf:protobuf-java:3.21.7 -> 3.21.12
|    |    |    +--- com.google.api.grpc:proto-google-common-protos:2.9.0 -> 2.14.2 (*)
|    |    |    +--- io.grpc:grpc-protobuf-lite:1.53.0
|    |    |    |    +--- io.grpc:grpc-api:1.53.0 (*)
|    |    |    |    +--- com.google.code.findbugs:jsr305:3.0.2
|    |    |    |    \--- com.google.guava:guava:31.1-android -> 31.1-jre (*)
|    |    |    \--- com.google.guava:guava:31.1-android -> 31.1-jre (*)
|    |    +--- io.grpc:grpc-protobuf-lite:1.53.0 (*)
|    |    +--- com.google.api:api-common:2.6.2
|    |    +--- com.google.auto.value:auto-value-annotations:1.10.1
|    |    +--- com.google.auto.value:auto-value:1.10.1
|    |    +--- javax.annotation:javax.annotation-api:1.3.2
|    |    +--- com.google.protobuf:protobuf-java:3.21.12
|    |    +--- com.google.api.grpc:proto-google-common-protos:2.14.2 (*)
|    |    +--- com.google.api.grpc:proto-google-cloud-trace-v1:2.12.0
|    |    +--- com.google.api.grpc:proto-google-cloud-trace-v2:2.12.0
|    |    |    +--- com.google.auto.value:auto-value-annotations:1.10.1
|    |    |    +--- com.google.auto.value:auto-value:1.10.1
|    |    |    +--- com.google.code.findbugs:jsr305:3.0.2
|    |    |    +--- javax.annotation:javax.annotation-api:1.3.2
|    |    |    +--- com.google.errorprone:error_prone_annotations:2.18.0
|    |    |    +--- com.google.guava:failureaccess:1.0.1
|    |    |    +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|    |    |    +--- org.checkerframework:checker-qual:3.31.0
|    |    |    \--- com.google.j2objc:j2objc-annotations:1.3
|    |    +--- com.google.guava:guava:31.1-jre (*)
|    |    +--- com.google.guava:failureaccess:1.0.1
|    |    +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|    |    +--- org.checkerframework:checker-qual:3.31.0
|    |    +--- com.google.j2objc:j2objc-annotations:1.3
|    |    +--- com.google.api:gax:2.23.2
|    |    +--- com.google.auth:google-auth-library-credentials:1.16.0
|    |    +--- io.opencensus:opencensus-api:0.31.1
|    |    +--- com.google.auth:google-auth-library-oauth2-http:1.16.0
|    |    +--- com.google.api:gax-grpc:2.23.2
|    |    +--- io.grpc:grpc-alts:1.53.0
|    |    +--- io.grpc:grpc-grpclb:1.53.0
|    |    +--- org.conscrypt:conscrypt-openjdk-uber:2.5.2
|    |    +--- io.grpc:grpc-auth:1.53.0
|    |    +--- io.grpc:grpc-netty-shaded:1.53.0
|    |    |    +--- com.google.guava:guava:31.1-android -> 31.1-jre (*)
|    |    |    +--- com.google.errorprone:error_prone_annotations:2.14.0 -> 2.18.0
|    |    |    +--- io.perfmark:perfmark-api:0.25.0 -> 0.26.0
|    |    |    \--- io.grpc:grpc-core:1.53.0
|    |    |         +--- io.grpc:grpc-api:1.53.0 (*)
|    |    |         +--- com.google.code.gson:gson:2.9.0 -> 2.10.1
|    |    |         +--- com.google.android:annotations:4.1.1.4
|    |    |         +--- org.codehaus.mojo:animal-sniffer-annotations:1.21 -> 1.22
|    |    |         +--- com.google.errorprone:error_prone_annotations:2.14.0 -> 2.18.0
|    |    |         +--- com.google.guava:guava:31.1-android -> 31.1-jre (*)
|    |    |         \--- io.perfmark:perfmark-api:0.25.0 -> 0.26.0
|    |    +--- io.perfmark:perfmark-api:0.26.0
|    |    +--- io.grpc:grpc-core:1.53.0 (*)
|    |    +--- com.google.android:annotations:4.1.1.4
|    |    +--- org.codehaus.mojo:animal-sniffer-annotations:1.22
|    |    +--- io.grpc:grpc-googleapis:1.53.0
|    |    +--- io.grpc:grpc-xds:1.53.0
|    |    +--- io.opencensus:opencensus-proto:0.2.0
|    |    +--- io.grpc:grpc-services:1.53.0
|    |    +--- com.google.re2j:re2j:1.6
|    |    +--- com.google.api:gax-httpjson:0.108.2
|    |    +--- com.google.code.gson:gson:2.10.1
|    |    +--- com.google.http-client:google-http-client:1.43.0
|    |    +--- org.apache.httpcomponents:httpclient:4.5.14
|    |    +--- commons-logging:commons-logging:1.2
|    |    +--- commons-codec:commons-codec:1.15
|    |    +--- org.apache.httpcomponents:httpcore:4.4.16
|    |    +--- io.opencensus:opencensus-contrib-http-util:0.31.1
|    |    +--- com.google.http-client:google-http-client-gson:1.43.0
|    |    +--- com.google.protobuf:protobuf-java-util:3.21.12
|    |    \--- org.threeten:threetenbp:1.6.5
|    \--- com.google.api.grpc:grpc-google-cloud-trace-v2:2.1.12 -> 2.12.0
|         +--- io.grpc:grpc-api:1.53.0 (*)
|         +--- io.grpc:grpc-context:1.53.0
|         +--- com.google.code.findbugs:jsr305:3.0.2
|         +--- com.google.errorprone:error_prone_annotations:2.18.0
|         +--- io.grpc:grpc-stub:1.53.0 (*)
|         +--- io.grpc:grpc-protobuf:1.53.0 (*)
|         +--- com.google.api.grpc:proto-google-common-protos:2.14.2 (*)
|         +--- io.grpc:grpc-protobuf-lite:1.53.0 (*)
|         +--- com.google.protobuf:protobuf-java:3.21.12
|         +--- com.google.api.grpc:proto-google-cloud-trace-v2:2.12.0 (*)
|         +--- com.google.api:api-common:2.6.2
|         +--- com.google.auto.value:auto-value-annotations:1.10.1
|         +--- com.google.auto.value:auto-value:1.10.1
|         +--- javax.annotation:javax.annotation-api:1.3.2
|         +--- com.google.guava:guava:31.1-jre (*)
|         +--- com.google.guava:failureaccess:1.0.1
|         +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|         +--- org.checkerframework:checker-qual:3.31.0
|         \--- com.google.j2objc:j2objc-annotations:1.3
+--- com.google.cloud.opentelemetry:exporter-auto:0.24.0-alpha // Shaded, so no transitive dependencies included 
+--- io.grpc:grpc-protobuf:1.52.1 -> 1.53.0 (*) // grpc libraries version updated
+--- io.grpc:grpc-stub:1.52.1 -> 1.53.0 (*)  // grpc libraries version udpated
+--- com.google.cloud:libraries-bom:26.10.0
|    +--- io.grpc:grpc-netty-shaded:1.53.0 (c)  // adding BOM for google libraries forces grpc version to be 1.53.0
|    +--- io.grpc:grpc-protobuf:1.53.0 (c)
|    +--- io.grpc:grpc-stub:1.53.0 (c)
|    +--- com.google.auto.value:auto-value-annotations:1.10.1 (c)
|    +--- com.google.cloud:google-cloud-core:2.12.0 (c)
|    +--- com.google.cloud:google-cloud-trace:2.12.0 (c)
|    +--- com.google.api.grpc:grpc-google-cloud-trace-v2:2.12.0 (c)
|    +--- com.google.guava:guava:31.1-jre (c)
|    +--- io.grpc:grpc-core:1.53.0 (c)
|    +--- io.grpc:grpc-api:1.53.0 (c)
|    +--- com.google.protobuf:protobuf-java:3.21.12 (c)
|    +--- com.google.api.grpc:proto-google-common-protos:2.14.2 (c)
|    +--- io.grpc:grpc-protobuf-lite:1.53.0 (c)
|    +--- com.google.api:gax:2.23.2 (c)
|    +--- io.grpc:grpc-context:1.53.0 (c)
|    +--- com.google.protobuf:protobuf-java-util:3.21.12 (c)
|    +--- com.google.code.gson:gson:2.10.1 (c)
|    +--- com.google.api.grpc:proto-google-iam-v1:1.9.2 (c)
|    +--- com.google.api:api-common:2.6.2 (c)
|    +--- com.google.auth:google-auth-library-credentials:1.16.0 (c)
|    +--- com.google.auth:google-auth-library-oauth2-http:1.16.0 (c)
|    +--- com.google.http-client:google-http-client:1.43.0 (c)
|    +--- com.google.http-client:google-http-client-gson:1.43.0 (c)
|    +--- com.google.api.grpc:proto-google-cloud-trace-v1:2.12.0 (c)
|    +--- com.google.api.grpc:proto-google-cloud-trace-v2:2.12.0 (c)
|    +--- com.google.api:gax-grpc:2.23.2 (c)
|    +--- io.grpc:grpc-alts:1.53.0 (c)
|    +--- io.grpc:grpc-grpclb:1.53.0 (c)
|    +--- io.grpc:grpc-auth:1.53.0 (c)
|    +--- io.grpc:grpc-googleapis:1.53.0 (c)
|    +--- io.grpc:grpc-xds:1.53.0 (c)
|    +--- io.grpc:grpc-services:1.53.0 (c)
|    \--- com.google.api:gax-httpjson:0.108.2 (c)
+--- io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha:1.23.0-alpha
|    +--- io.opentelemetry:opentelemetry-bom:1.23.1
|    |    +--- io.opentelemetry:opentelemetry-api:1.23.1 (c)
|    |    +--- io.opentelemetry:opentelemetry-sdk:1.23.1 (c)
|    |    +--- io.opentelemetry:opentelemetry-context:1.23.1 (c)
|    |    +--- io.opentelemetry:opentelemetry-sdk-common:1.23.1 (c)
|    |    +--- io.opentelemetry:opentelemetry-sdk-trace:1.23.1 (c)
|    |    \--- io.opentelemetry:opentelemetry-sdk-metrics:1.23.1 (c)
|    +--- io.opentelemetry:opentelemetry-bom-alpha:1.23.1-alpha
|    |    +--- io.opentelemetry:opentelemetry-bom:1.23.1 (*)
|    |    +--- io.opentelemetry:opentelemetry-semconv:1.23.1-alpha (c)
|    |    +--- io.opentelemetry:opentelemetry-sdk-logs:1.23.1-alpha (c)
|    |    +--- io.opentelemetry:opentelemetry-api-logs:1.23.1-alpha (c)
|    |    \--- io.opentelemetry:opentelemetry-api-events:1.23.1-alpha (c)
|    \--- io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom:1.23.0
|         \--- io.opentelemetry:opentelemetry-bom:1.23.1 (*)
\--- io.grpc:grpc-netty-shaded:1.52.1 -> 1.53.0 (*) // grpc libraries version updated

(c) - dependency constraint
(*) - dependencies omitted (listed previously)
```
As seen here, BOM adds dependency constraints (*indicated by c above*) which are honored by the transitive dependencies, since `exporter-auto` does not have any transitive dependencies, I do not think using dependency version declared in BOMs would be affected by `exporter-auto`. 

Also, regarding #184, the issue suggests that the conflict was regarding `exporter-trace`, which is provided in a non-shaded variant and therefore it pulls in transitive dependencies (*visible in the above output*) which may have caused conflicts.

Let me know if this makes things clear.

**Comment by psx95** [comment]
> I think what is happening is that Maven looks only at <dependencies> section of exporter-auto and doesn't see any dependencies to overload with BOM.

Regarding the part about Maven not overloading dependencies, I think you are right, since there are no transitive dependencies, they will not be overloaded and this is the expected behavior. 
In some cases, depending on your application code, this might lead to issues as described in #161 .

**Comment by ViliusS** [comment]
> @ViliusS, I am looking into this. I will try it out with Maven once as you suggested, but as per your question on "how and what dependencies are resolved when using exporter-auto with 3rd party BOMs", I think `exporter-auto` should be treated as an artifact which does not have any transitive dependency. Since currently its provided as a shaded artifact it bundles every single dependency in itself.
> 
> That being said, I noticed that we are not moving the classes to a shaded package for `exporter-auto` right now which can still lead to issues described in #161 (_Is this something you are running into ?_).

It's hard to say, I don't fully understand how shared dependencies work. We can wait for the fix in #161 to see if that changes anything for my case.

**Comment by psx95** [comment]
Sounds good. In the mean time, I will close this issue. 

I also found [this article](https://medium.com/@akhaku/java-class-shadowing-and-shading-9439b0eacb13) to be helpful, if you wish to read. You can skip to "Shading in Java" part to get a gist.
